x-pack/plugins/ml/server/saved_objects/checks.ts

```diff
@@ -0,0 +1,271 @@
+      savedObjects.filter(({ type }) => type === 'data-frame-analytics').map(({ jobId }) => jobId)
+    );
+
+    const anomalyDetectors = adJobs.jobs
+      .filter(({ job_id: jobId }) => {
+        // only list jobs which are in the current space (adObjectIds)
+        // or are not in any spaces (nonSpaceADObjectIds)
+        return adObjectIds.has(jobId) === true || nonSpaceADObjectIds.has(jobId) === false;
+      })
+      .map(({ job_id: jobId }) => {
+        const datafeedId = datafeeds.datafeeds.find((df) => df.job_id === jobId)?.datafeed_id;
+        return {
+          jobId,
+          datafeedId: datafeedId ?? null,
+          checks: {
+            savedObjectExits: nonSpaceADObjectIds.has(jobId),
+          },
+        };
+      });
+
+    const dataFrameAnalytics = dfaJobs.data_frame_analytics
+      .filter(({ id: jobId }) => {
+        // only list jobs which are in the current space (dfaObjectIds)
+        // or are not in any spaces (nonSpaceDFAObjectIds)
+        return dfaObjectIds.has(jobId) === true || nonSpaceDFAObjectIds.has(jobId) === false;
+      })
+      .map(({ id: jobId }) => {
+        return {
+          jobId,
+          checks: {
+            savedObjectExits: nonSpaceDFAObjectIds.has(jobId),
+          },
+        };
+      });
+
+    return {
+      savedObjects: {
+        anomalyDetectors: savedObjects.filter(({ type }) => type === 'anomaly-detector'),
+        dataFrameAnalytics: savedObjects.filter(({ type }) => type === 'data-frame-analytics'),
+      },
+      jobs: {
+        anomalyDetectors,
+        dataFrameAnalytics,
+      },
+    };
+  }
+
+  async function repairJobs(simulate: boolean = false) {
+    const results: {
+      savedObjectsCreated: string[];
+      savedObjectsDeleted: string[];
+      datafeedsAdded: string[];
+      datafeedsRemoved: string[];
+    } = {
+      savedObjectsCreated: [],
+      savedObjectsDeleted: [],
+      datafeedsAdded: [],
+      datafeedsRemoved: [],
+    };
+
+    const { body: datafeeds } = await client.asInternalUser.ml.getDatafeeds<{
+      datafeeds: Datafeed[];
+    }>();
+
+    const tasks: Array<() => Promise<void>> = [];
+
+    const status = await checkStatus();
+    for (const job of status.jobs.anomalyDetectors) {
+      if (job.checks.savedObjectExits === false) {
+        results.savedObjectsCreated.push(job.jobId);
+        if (simulate === false) {
+          const jobId = job.jobId;
+          const datafeedId = job.datafeedId;
+          tasks.push(async () => {
+            await jobSavedObjectService.createAnomalyDetectionJob(jobId);
+            if (datafeedId !== null) {
+              //
+              await jobSavedObjectService.addDatafeed(datafeedId, jobId);
+            }
+          });
+        }
+      }
+    }
+    for (const job of status.jobs.dataFrameAnalytics) {
+      if (job.checks.savedObjectExits === false) {
+        results.savedObjectsCreated.push(job.jobId);
+        if (simulate === false) {
+          //
```

Comment:
Did you intend to add comments in this file? There are a lot of `//` with no text.